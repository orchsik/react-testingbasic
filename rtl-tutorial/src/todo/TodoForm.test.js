import { fireEvent, render } from "@testing-library/react";
import TodoForm from "./TodoForm";

const INPUT_PLACE_HOLDER = "할 일을 입력하세요";
const BUTTON_TEXT = "등록";

const setup = (props = {}) => {
  const utils = render(<TodoForm {...props} />);
  const { getByText, getByPlaceholderText } = utils;
  const input = getByPlaceholderText("할 일을 입력하세요");
  const button = getByText("등록");
  return { ...utils, input, button };
};

describe("<TodoForm />", () => {
  it("matches sanpshot", () => {
    const { container } = setup();
    expect(container).toMatchSnapshot();
  });

  it("has input and a button", () => {
    const { getByText, getByPlaceholderText } = render(<TodoForm />);
    getByPlaceholderText(INPUT_PLACE_HOLDER);
    getByText(BUTTON_TEXT);
  });

  it("changes input", () => {
    const { input } = setup();
    fireEvent.change(input, { target: { value: "TDD 배우기" } });
    expect(input).toHaveAttribute("value", "TDD 배우기");
    expect(input.value).toBe("TDD 배우기");
  });

  it("calls onInsert and clears input", () => {
    const onInsert = jest.fn();
    const { input, button } = setup({ onInsert });

    fireEvent.change(input, { target: { value: "TDD 배우기" } });
    fireEvent.click(button);
    expect(onInsert).toBeCalledWith("TDD 배우기");
    expect(input).toHaveAttribute("value", "");
  });
});
