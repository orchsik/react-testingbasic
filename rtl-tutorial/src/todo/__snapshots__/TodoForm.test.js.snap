// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TodoForm /> matches sanpshot 1`] = `
<div>
  <form>
    <input
      placeholder="할 일을 입력하세요"
      value=""
    />
    <button
      type="submit"
    >
      등록
    </button>
  </form>
</div>
`;
